signer/src/error.rs

```diff
@@ -171,13 +171,21 @@ pub enum Error {
     #[error("unexpected public key from signature. key {0}; digest: {1}")]
     UnknownPublicKey(secp256k1::PublicKey, secp256k1::Message),
 
-    /// WSTS error
+    /// WSTS error.
     #[error("WSTS error: {0}")]
     Wsts(#[source] wsts::state_machine::signer::Error),
 
     /// No chain tip found.
     #[error("no bitcoin chain tip")]
     NoChainTip,
+
+    /// Bitcoin address parse error
+    #[error("bitcoin address parse error")]
+    BitcoinAddressParseError(#[source] bitcoin::address::ParseError),
```

Comment:
nit: This should be `BitcoinAddressParse`, the added `Error` is redundant. But maybe we should rename it to `BitcoinNetworkValidation`? Or not, the `Parse` part is okay as well since it's conceptually part of string parsing.

Reply:
Ah how could I miss this. Fixed!